lib/record_store/provider/ns1.rb

```diff
@@ -219,6 +220,35 @@ def symbolize_keys(hash)
       def secrets
         super.fetch('ns1')
       end
+
+      # Patch the method which retrieves headers for API rate limit dynamically
+      module NS1::Transport
+        refine NetHttp do
+          def process_response(response)
+            ratelimit = ['remaining', 'period']
+              .map { |k| [k, response["x-ratelimit-#{k}"].to_i] }.to_h
+
+            ratelimit_remaining = ratelimit["remaining"]
```

Comment:
Using the map _then_ doing this seems overkill. Just pull the data once directly from the headers!